### PR TITLE
Add shim for Python's gRPC core on Windows

### PR DESCRIPTION
**Pull request by soltanmm**
cc @nathanielmanistaatgoogle @ctiller

### CODE REVIEW

File: include/grpc/impl/codegen/port_platform.h [hunk 1]
```diff
@@ -34,6 +34,21 @@
...
+typedef __int32 int32_t;
+typedef __int64 int64_t;
+typedef unsigned __int8 uint8_t;
+typedef unsigned __int16 uint16_t;
+typedef unsigned __int32 uint32_t;
+typedef unsigned __int64 uint64_t;
+#else
+#include <stdint.h>
+#endif /* _MSC_VER < 1700 */
+#endif /* _MSC_VER */
```
Comment (nicolasnoble) [inline include/grpc/impl/codegen/port_platform.h]: That logic will break non windows builds. We still need to include stdint.h everywhere else.

File: include/grpc/impl/codegen/compression_types.h [hunk 2]
```diff
@@ -34,7 +34,7 @@
 #ifndef GRPC_IMPL_CODEGEN_COMPRESSION_TYPES_H
 #define GRPC_IMPL_CODEGEN_COMPRESSION_TYPES_H
 
-#include <stdint.h>
+#include <grpc/impl/codegen/port_platform.h>
```
Comment (nicolasnoble) [inline include/grpc/impl/codegen/compression_types.h]: Please use <grpc/support/port_platform.h>

File: include/grpc/impl/codegen/propagation_bits.h [hunk 3]
```diff
@@ -34,7 +34,7 @@
 #ifndef GRPC_IMPL_CODEGEN_H
 #define GRPC_IMPL_CODEGEN_H
 
-#include <stdint.h>
+#include <grpc/impl/codegen/port_platform.h>
```
Comment (nicolasnoble) [inline include/grpc/impl/codegen/propagation_bits.h]: Please use grpc/support/port_platform.h

File: src/python/grpcio/grpc/_cython/_cygrpc/grpc.pxi [hunk 4]
```diff
@@ -28,15 +28,20 @@
...
 cimport libc.time
-from libc.stdint cimport int64_t, uint32_t, int32_t
 
 
-cdef extern from "grpc/support/alloc.h":
+cdef extern from "grpc/_cython/loader.h":
+
+  ctypedef int int32_t
+  ctypedef unsigned uint32_t
+  ctypedef long int64_t
```
Comment (nicolasnoble) [inline src/python/grpcio/grpc/_cython/_cygrpc/grpc.pxi]: Is this a cython thing? Because "long" is 32 bits on Windows.

**Comment by nicolasnoble** [comment]
You have a bunch of temporary files in that pr. Also the binary dlls. If there's commits referencing these, the download size for the git clone will be forever affected, so please remove all the binary files and squash the commits.

File: include/grpc/impl/codegen/port_platform.h (hunk 1 above)
Comment (soltanmm) [inline include/grpc/impl/codegen/port_platform.h]: oops, yep.

File: src/python/grpcio/grpc/_cython/_cygrpc/grpc.pxi (hunk 4 above)
Comment (soltanmm) [inline src/python/grpcio/grpc/_cython/_cygrpc/grpc.pxi]: It's a Cython thing in the sense that the type doesn't need to be the exact type, it just needs to be a type supporting the same interface as the type actually being used that Cython is aware of elsewhere (so the `ctypedef` keyword is a kind of misnomer in this scenario).

**Comment by soltanmm** [comment]
^ addressed

**Comment by nicolasnoble** [comment]
Something broke for all languages in that PR. I'll look into that, possibly forking that PR.

**Comment by nicolasnoble** [comment]
#5064 replaces this one. Took me a while to realize what was broken - you moved the inclusion of stdint.h to the top of port_platform.h. Why did you move it ? :) It was down here for a reason.

**Comment by nicolasnoble** [comment]
Also in #5064 I am properly cleaning up the remaining binary files.

**Pull request closed without merging**